aws/aws-sdk-net
# exclude sns from Unity packages

## Description
With this change, plugin binary aws-unity-sns-gcm-helper.jar will not be copied to a unity package , except to SimpleNotificationService. The plugin binaries which are copied to an aws unity package can be configured service by service.

binaries under Plugins folder are removed. Instead, Common and Default folders are created under Plugins folder. 
The Common folder contains all binaries which are required by all services. 

If a service has special requirements, a folder with the service's name should be created under Plugins folder, all binaries for this service other than binaries in Common folder should be put to this folder.

The Default folder contains binaries which are required by a service which does not have a special requirement.  The binaries in Common folder should be included in this folder. 



## Motivation and Context
Customer is using AWS SDK for Unity[1] just to do the s3 GET request.
Issue:
- He sees that the when he using AWS SDK and builds the APK the Android studio analyze the apk and sees the GoogleCloudMessaging APK being called.

How to reproduce the issue and verify this behavior:
- Create a Unity project.
- Add the aws sdk for s3 and build the apk.
- then, open the apk with Android studio, you could see all files in the apk. click on classes.dex.
- down, in the class windows, you could see this: com.google.android.gms.gcm.GoogleCloudMessaging.  


## Testing

With the friendly CustomTasks.dll, run command msbuild build.proj /t:unity-pack  to generate the  aws sdk unity packages
 

with the unity packages generated: 
Case 1: 
1.  Create a Unity project.
2.  Select Unity menu,  Assets->Import Package->Custom Package, select AWSSDK.SimpleNotificationService.3.3.1.2.unitypackage from \aws-sdk-net\Deployment\unity\unitypackages. 
3. In the Import Unity Package windows. make sure plugins contains aws-unity-sns-gcm-helper.jar.
then select all components in the windows
4.  Make sure folder Assets\Plugins of the project contains correct binaries(aws-unity-sns-gcm-helper.jar should be present) and has a correct folder structure. 
5. build a android apk from the project. 
6. open the apk from android studio,  in the class windows of android studio,   com.google.android.gms.gcm.GoogleCloudMessaging should be seen. 

case 2
For each none SimpleNotificationService  unitypackage
1.  Create a Unity project.
2.  Select Unity menu  Assets->Import Package->Custom Package , select the none SimpleNotificationService .unitypackage from \aws-sdk-net\Deployment\unity\unitypackages. 
3. In the Import Unity Package windows. make sure plugins does not contain aws-unity-sns-gcm-helper.jar.   then  select all components in the windows
4.  Make sure folder Assets\Plugins of the project contains correct binaries (aws-unity-sns-gcm-helper.jar should not be present) and has a correct folder structure. 
5. build a android apk from the project. 
6. open the apk from android studio,  in the class windows of android studio,   com.google.android.gms.gcm.GoogleCloudMessaging should not be present. 




 

## Types of changes
 

## Checklist



## License
